cub/device/dispatch/dispatch_select_if.cuh

```diff
@@ -167,7 +167,308 @@ __launch_bounds__(int(ChainedPolicyT::ActivePolicy::SelectIfPolicyT::BLOCK_THREA
 namespace detail 
 {
 
-template <class InputT, bool MayAlias>
+namespace select 
+{
+
+template <class InputT,
+          bool Flagged,
```

Comment:
Food for thought, with no change requested:

I personally really dislike boolean template arguments. It is really easy to mess up 
`sm90_tuning<Input, false, false, 4, true, 1>`

On the other hand it is really hard to mess up
 `sm90_tuning<Input, Flagged::No, KeepRejects::No, 4, PrimitiveInput::Yes, 1>`
, when utilizing a lot of `enum class Flagged : bool { No, Yes };`

I appreciate, that it adds a considerable amount of code in the implementation and is totally out of scope of this PR. Just wanting to plant a seed

Reply:
Agreed! Will implement as part of this PR.